de.fu_berlin.inf.dpp.core/test/junit/de/fu_berlin/inf/dpp/versioning/JGitServiceTest.java

```diff
@@ -0,0 +1,69 @@
+package de.fu_berlin.inf.dpp.versioning;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.Status;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class JGitServiceTest {
+
+    public enum RepositoryOwner {
```

Comment:
The concept of Host and Client is not really clear.
You mean that the host is the User (e.g. Alice) that wants to share a commit with the client (e.g. Bob) or vice versa?
I don't know whether this terminology is useful.

See my comment in line 43